File: backend/planta-smart-4.0/src/main/java/com/smart/appsa/mapper/EstoqueMapper.java
package com.smart.appsa.mapper;

import com.smart.appsa.dto.request.EstoqueRequestDTO;
import com.smart.appsa.dto.response.EstoqueResponseDTO;
import com.smart.appsa.model.Estoque;

public class EstoqueMapper {

    public static EstoqueResponseDTO mapDTO(Estoque estoque) {
        return EstoqueResponseDTO.builder()
            .id(estoque.getId())
            .posicaoFisica(estoque.getPosicaoFisica())
            .corEstoque(estoque.getCorEstoque())
            .build();
    }

    public static Estoque mapEntityByResponseDTO(EstoqueResponseDTO dto) {
        return Estoque.builder()
            .id(dto.id())
            .posicaoFisica(dto.posicaoFisica())
            .corEstoque(dto.corEstoque())
            .build();
    }

    public static Estoque mapEntityByRequestDTO(EstoqueRequestDTO dto) {
        return Estoque.builder()
            .id(dto.id())
            .posicaoFisica(dto.posicaoFisica())
            .corEstoque(dto.corEstoque())
            .build();
    }

    public static EstoqueRequestDTO mapRequestDTO(Estoque estoque) {
        return EstoqueRequestDTO.builder()
            .id(estoque.getId())
            .posicaoFisica(estoque.getPosicaoFisica())
            .corEstoque(estoque.getCorEstoque())
            .build();
    }
}

File: frontend/planta-smart-4.0/src/infrastructure/mappers/EstoqueMapper.ts
import { EstoqueRequestDTO } from '@dtos/request/EstoqueRequestDTO';
import { EstoqueResponseDTO } from '@dtos/response/EstoqueResponseDTO';
import { Estoque } from '@entities/Estoque';
import { CorEstoque } from '@enums/CorEstoque';

const corEstoqueMap: Record<string, CorEstoque> = {
  VAZIO: CorEstoque.Vazio,
  PRETO: CorEstoque.Preto,
  VERMELHO: CorEstoque.Vermelho,
  AZUL: CorEstoque.Azul,
};

export const EstoqueMapper = {
  mapEntityByResponseDTO(dto: EstoqueResponseDTO): Estoque {
    return {
      id: dto.id,
      posicaoFisica: dto.posicaoFisica,
      cor: corEstoqueMap[dto.corEstoque.toUpperCase()],
    };
  },

  mapEntitiesByResponsesDTOs(dtos: EstoqueResponseDTO[]): Estoque[] {
    const responses: Estoque[] = [];

    dtos.forEach((dto) => {
      responses.push(this.mapEntityByResponseDTO(dto));
    });

    return responses;
  },

  mapRequestDTOByEntity(entity: Estoque): EstoqueRequestDTO {
    return {
      id: entity.id,
      posicaoFisica: entity.posicaoFisica,
      corEstoque: entity.cor,
    };
  },

  mapRequestsDTOByEntities(entities: Estoque[]): EstoqueRequestDTO[] {
    const requests: EstoqueRequestDTO[] = [];

    entities.forEach((entity) => {
      requests.push(this.mapRequestDTOByEntity(entity));
    });

    return requests;
  },
};
